GVFS/GVFS.Virtualization/FileSystem/FileSystemVirtualizer.cs

```diff
@@ -269,17 +269,24 @@ protected void OnHardLinkCreated(string relativeExistingFilePath, string relativ
         {
             try
             {
-                if (!string.IsNullOrEmpty(relativeNewLinkPath))
+                if (!string.IsNullOrEmpty(relativeNewLinkPath) || !string.IsNullOrEmpty(relativeExistingFilePath))
                 {
-                    bool pathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeNewLinkPath);
+                    bool newLinkPathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeNewLinkPath);
+                    bool existingPathInDotGit = FileSystemCallbacks.IsPathInsideDotGit(relativeExistingFilePath);
 
-                    if (pathInDotGit)
+                    if (newLinkPathInDotGit)
                     {
                         this.OnDotGitFileOrFolderChanged(relativeNewLinkPath);
                     }
-                    else
+
+                    if (existingPathInDotGit)
+                    {
+                        this.OnDotGitFileOrFolderChanged(relativeExistingFilePath);

```

Comment:
I can't think of a scenario where `OnDotGitFileOrFolderChanged` would need to be called for `relativeExistingFilePath`, did you have one in mind?

Reply:
All of the git folder cases are a bit strange to me, but maybe you can enlighten me here :-)
I'll followup offline

Reply:
Summary of the offline coversation (as far as I understand)

The .git folder tracking is mainly for timestamps / seeing the if the git repo changed.  We care about new links here (an update could be done via a hardlink / file replacement).  We shouldn't track the existing file for a hardlink.